### ISSUE TITLE
[package] m4/1.4.19: unable to load recipe

### ISSUE THREAD

**Issue by bowb** [conan-io/conan-center-index#13224]
### Description

Unable to iinstall m4/1.4.19 with conan 1.51.3

### Package and Environment Details

* Package Name/Version: **m4/1.4.19**
* Operating System+version: **Linux Ubuntu 20.04**
* Compiler+version: **GCC 10.3**
* Docker image: **conanio/gcc8**
* Conan version: **conan 1..51.3**
* Python version: **Python 3.8**

### Conan profile

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=10
compiler.libcxx=libstdc++11
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

conan install -r conancenter m4/1.4.19@  -pr:b=default -pr:h=default

### Logs

<details><summary>Click to expand log</summary>

```
ERROR: m4/1.4.19: Cannot load recipe.
Error loading conanfile at '/root/.conan/data/m4/1.4.19/_/_/export/conanfile.py': Unable to load conanfile in /root/.conan/data/m4/1.4.19/_/_/export/conanfile.py
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/root/.conan/data/m4/1.4.19/_/_/export/conanfile.py", line 3, in <module>
    from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir, save
ImportError: cannot import name 'export_conandata_patches' from 'conan.tools.files' (/usr/local/lib/python3.8/dist-packages/conan/tools/files/__init__.py)
```

</details>

**Comment by ericLemanissier** [comment]
you need to upgrade conan, cf https://github.com/conan-io/conan-center-index/blob/master/recipes/m4/all/conanfile.py#L10
The error message is unfortunately not pointing into this direction :/

**Comment by uilianries** [comment]
@ericLemanissier IIRC 1.52.0 (or 1.53.0) will  read required_conan_version first to avoid this kind of error

**Comment by ericLemanissier** [comment]
this one can probably be closed,now that https://github.com/conan-io/conan/pull/11908 has been released with conan 1.52.0

**Comment by czoido** [comment]
Closing this as looks fixed, please feel free to reopen if you have more issues.
